### PR TITLE
close #1097 add segment bitwise to homepage_sections table

### LINKED ISSUE #1097: Add segment column to homepage_section table
PR #1110

### PR DESCRIPTION
**Pull request by vaneath**
[Screencast from 2024-02-15 15-18-25.webm](https://github.com/channainfo/commissioner/assets/109834020/bead12fc-f132-4918-839a-34073240f232)

### CODE REVIEW

**Comment by channainfo** [comment]
@vaneath After a new section is created, redirect to edit instead because it makes more sense to add Relatable items. The new section is useless without any relatables. Also, add a counter_cache to the section relatable. Display the number of section relatable count in the table. Make sure we only return section where it section relatable count is great than 1 ( using the counter_cache)

**Comment by vaneath** [comment]
> @vaneath After a new section is created, redirect to edit instead because it makes more sense to add Relatable items. The new section is useless without any relatables. Also, add a counter_cache to the section relatable. Display the number of section relatable count in the table. Make sure we only return section where it section relatable count is great than 1 ( using the counter_cache)

noted bong, I will check on it

File: app/models/concerns/spree_cm_commissioner/homepage_section_bitwise.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module SpreeCmCommissioner
+  module HomepageSectionBitwise
+    extend ActiveSupport::Concern
+
+    BIT_SEGMENT = {
+      general: 0b00001,
```
Comment (theachoem) [inline app/models/concerns/spree_cm_commissioner/homepage_section_bitwise.rb:6]: Let add tour & accommodation as well. `general, ticket, tour, accommodation`

File: app/helpers/spree_cm_commissioner/admin/homepage_segment_helper.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module SpreeCmCommissioner
+  module Admin
+    module HomepageSegmentHelper
+      def calculate_segment_value(params)
+        segment_params = params.slice(:general, :ticket)
```
Comment (theachoem) [inline app/helpers/spree_cm_commissioner/admin/homepage_segment_helper.rb:5]: params.slice(*SpreeCmCommissioner::HomepageSectionBitwise:: BIT_SEGMENT.keys)

**Review by theachoem — CHANGES_REQUESTED** [review]

File: app/models/spree_cm_commissioner/homepage_section.rb [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
-    attribute :description
-    attribute :active
-    attribute :position
+    attribute :segment, :integer, default: 0
 
     has_many :homepage_section_relatables, -> { active }, inverse_of: :homepage_section, dependent: :destroy
 
     scope :active, -> { where(active: true).order(:position) }
+
+    def self.badge_class_for_segment(segment_name)
```
Comment (theachoem) [inline app/models/spree_cm_commissioner/homepage_section.rb:13]: Move this to controller helper

**Review by theachoem — CHANGES_REQUESTED** [review]

**Pull request merged**